src/librustc_metadata/creader.rs

```diff
@@ -367,16 +367,31 @@ impl<'a> CrateLoader<'a> {
         let cmeta = self.cstore.get_crate_data(cnum);
         let mut old_extern_crate = cmeta.extern_crate.borrow_mut();
 
+        fn path_len_reverse(src: ExternCrateSource) -> cmp::Reverse<usize> {
+            cmp::Reverse(match src {
+                ExternCrateSource::Extern { path_len, .. } |
+                ExternCrateSource::Use { path_len } => path_len,
+                _ => usize::max_value(),
```

Comment:
It looks like it'd be more convenient to move `path_len` into `ExternCrate` and initialize it to `usize::max_value()` for `ExternCrateSource::Path`.
(Maybe some better heuristic than `max_value` could be found later).

Reply:
Done.